### PR TITLE
Add rxnorm

### PR DESCRIPTION
**Pull request by jsstevenson**

### CODE REVIEW

File: src/bioversions/sources/rxnorm.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "RxNormGetter",
+]
+
+URL = "https://www.nlm.nih.gov/research/umls/rxnorm/docs/rxnormfiles.html"
+
+
+class RxNormGetter(Getter):
+    """A getter for RxNorm."""
+
+    bioregistry_id = "rxnorm"
```
Comment (jsstevenson) [inline src/bioversions/sources/rxnorm.py:18]: RxNorm is unfortunately not registered with identifiers.org or the BioRegistry -- I assumed it's tentatively fine to supply a reasonable value here?

Comment (cthoyt) [reply src/bioversions/sources/rxnorm.py:19]: http://bioregistry.io/registry/rxnorm should be what you're looking for

Comment (jsstevenson) [reply src/bioversions/sources/rxnorm.py:19]: Ah, great -- not sure how I missed that... very easily-findable entry.

File: src/bioversions/sources/rxnorm.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    version_type = VersionType.date
+    date_fmt = "%m%d%Y"
+
+    def get(self) -> str:
+        """Get the latest BioGRID version number."""
+        soup = get_soup(URL)
+        raw_version = soup.find("th", {"class": "current"}).contents[2]
+        raw_fmt = "%B %d, %Y"
+        dt = datetime.strptime(raw_version, raw_fmt)
+        version = datetime.strftime(dt, "%m%d%Y")
```
Comment (cthoyt) [inline src/bioversions/sources/rxnorm.py:31]: why did you choose to use this format? YYYY-MM-DD is much better (for example, it's sortable)

Comment (jsstevenson) [reply src/bioversions/sources/rxnorm.py:31]: Agreed! I wasn't sure whether I was constrained to matching the homepage_fmt string. Is something like this possible?

Comment (cthoyt) [reply src/bioversions/sources/rxnorm.py:31]: I'm a bit busy at the moment to grok this, but check this example https://github.com/biopragmatics/bioversions/blob/main/src/bioversions/sources/drugbank.py that shows where I had to put a transformation on the version before I formatted it in there

**Review by cthoyt — APPROVED** [review]
@jsstevenson everything looks good! Thanks for addressing the CI feedback. I made a few typo fixes and changed the getter to directly return the datetime.datetime object - this is shared functionality across a few other getters and makes it less cluttered to format into YYYY-MM-DD inside this class. I also enabled MyPy type checking in the meantime to help us write better code in the future :)

**Pull request merged**